corehq/apps/events/tasks.py

```diff
@@ -71,7 +71,7 @@ def get_user_attendee_cases_on_domain(domain):
 
 
 def get_case_block_for_user(user, owner_id, attendee_case_type):
-    case_name = ' '.join((user.first_name, user.last_name))
+    case_name = user.username
```

Comment:
Yes, because attendees that you create manually (i.e. who is not also a mobile worker) might also not satisfy the uniqueness criteria.

Reply:
Yeah as Charl mentioned. 

> but rather use the username from the start
This only pertains to the mobile workers. The `get_case_block_for_user` method is used to get a case block for a commcare user, which is a mobile user. (Confusingly, a couch user is either a web or mobile user)

Reply:
Ah that makes sense. So would attendance workers that are made manually still use a first name + last name, or would they also be given a username?

Reply:
The former for now, still using the good ol' first and last name

Reply:
What if, just to keep things a bit more consistent for now, we have a username property on the Attendee model that returns first name + last name? Then we can just fetch the username property, regardless if its a mobile worker or manual attendee. :thinking:

Reply:
> Then we can just fetch the username property

Technically it is being done following this principle, just in another way. We don't read the [content](https://github.com/dimagi/commcare-hq/blob/f048e90b961574a6d7b55caeea6d7640bfd3d4e0/corehq/apps/events/views.py#L372-L377) that is shown in the attendee table from a case property, but rather from the case name. What this PR's change is doing is to set the case name of mobile workers to the username. This way there's no manual check to return a username if it is a mobile worker, and a name+lastname otherwise. It just returns the case name

Reply:
Ah okay, that definitely clears up some of the confusion I had. In that case, the PR looks good. :+1: